### PR TITLE
[PAGOPA-627] get call: refactoring from httpTrigger to spring rest call

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- added get rest call

#### Motivation and Context

- a call to trigger the import CDI

#### How Has This Been Tested?

- manual and junit test

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/test/java/it/gov/pagopa/afm/utils/service/CDIServiceTest.java [hunk 1]
```diff
@@ -14,18 +14,27 @@
...
     private CDICollectionRepository cdisRepository;
+	@Mock
+	private MarketPlaceClient marketPlaceClient;
 	
 	@Test
 	void deleteTest() {
-		CDIService cdiService = new CDIService(cdisRepository);
+		CDIService cdiService = new CDIService(cdisRepository, marketPlaceClient);
 		cdiService.deleteCDI(CDI.builder().build());
 		assertTrue(true);
```
Comment (jacopocarlini) [inline src/test/java/it/gov/pagopa/afm/utils/service/CDIServiceTest.java:24]: Can we do more relevant assertions?

File: src/main/java/it/gov/pagopa/afm/utils/service/CDIService.java [hunk 2]
```diff
@@ -1,25 +1,200 @@
...
     public void deleteCDI(CDI cdiEntity) {
     	cdisRepository.delete(cdiEntity);	
     } 
+    
+    public List<CDI> getWorkableCDIs() {
+    	return cdisRepository.getWorkableCDIs();
+    }
+    
+    public List<BundleResponse> syncCDI() {
+    	return this.turnCDIToBundles(this.getWorkableCDIs());
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/afm/utils/service/CDIService.java:60]:
```suggestion
    	return this.turnCDIToBundles(cdisRepository.getWorkableCDIs());
```

File: src/main/java/it/gov/pagopa/afm/utils/service/CDIService.java [hunk 3]
```diff
@@ -1,25 +1,200 @@
...
         return cdisRepository.save(cdiEntity);
     }
     
     public void deleteCDI(CDI cdiEntity) {
     	cdisRepository.delete(cdiEntity);	
     } 
+    
+    public List<CDI> getWorkableCDIs() {
+    	return cdisRepository.getWorkableCDIs();
+    }
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/afm/utils/service/CDIService.java:57]:
```suggestion
```

**Review by jacopocarlini — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/afm/utils/ImportCDIFunction.java [hunk 4]
```diff
@@ -1,189 +1,34 @@
...
 
 @Component
-@Slf4j
 @NoArgsConstructor
 @AllArgsConstructor
-public class ImportCDIFunction  implements Function<Mono<CDIWrapper>, Mono<List<BundleResponse>>>{
-
-	@Autowired(required=false)
-	private MarketPlaceClient marketPlaceClient;
+public class ImportCDIFunction implements Function<Mono<CDIWrapper>, Mono<List<BundleResponse>>> {
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/afm/utils/ImportCDIFunction.java:21]: is this class used at the moment?

File: openapi/openapi.json [hunk 5]
```diff
@@ -13,14 +13,14 @@
     }
   ],
   "paths": {
-    "/cdis/sync": {
+    "/api/cdis/sync": {
```
[the remark this answers is not in the document]
Comment (alessio-acitelli) [reply openapi/openapi.json]: removed httptrigger /api/cdis/sync

File: src/main/java/it/gov/pagopa/afm/utils/controller/ImportCDIController.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+@RestController
+public class ImportCDIController {
+	
+	@Autowired
+	private CDIService cdiService;
+	
+	@Operation(summary = "API to trigger the import of the CDIs and convert to bundles.", security = {@SecurityRequirement(name = "ApiKey")}, operationId = "syncCDI", tags = {"Import CDI rest API"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Obtained bundle list.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, array = @ArraySchema(schema = @Schema(name = "BundleResponse", implementation = BundleResponse.class)))),
+            @ApiResponse(responseCode = "500", description = "Service unavailable.", content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE, schema = @Schema(implementation = ProblemJson.class)))})
```
[the remark this answers is not in the document]
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/afm/utils/controller/ImportCDIController.java:35]: added 401, 403 and 429 status code

File: src/main/java/it/gov/pagopa/afm/utils/service/CDIService.java (hunk 2 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/afm/utils/service/CDIService.java]: done

File: src/test/java/it/gov/pagopa/afm/utils/service/CDIServiceTest.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/test/java/it/gov/pagopa/afm/utils/service/CDIServiceTest.java]: done

File: src/main/java/it/gov/pagopa/afm/utils/controller/ImportCDIController.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import it.gov.pagopa.afm.utils.model.ProblemJson;
+import it.gov.pagopa.afm.utils.model.bundle.BundleResponse;
+import it.gov.pagopa.afm.utils.service.CDIService;
+
+@RestController
```
[the remark this answers is not in the document]
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/afm/utils/controller/ImportCDIController.java:24]: done

**Review by jacopocarlini — APPROVED** [review]
Good Job 🥇

**Pull request merged**